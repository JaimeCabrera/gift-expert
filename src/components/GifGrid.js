import React from 'react'
import PropTypes from 'prop-types'
import GifGridItem from './GifGridItem';
import { useFetchGifs } from '../hooks/useFetchGifs';

const GifGrid = ({ category }) => {

   // para renombrar en la desetructuracion name:newname
   const {data:images, loading} = useFetchGifs(category);
   // const [images, setImages] = useState([]);

   // useEffect(() => {
   //    getGifs(category).then(gifs => setImages(gifs));
   // }, [category])

   // recibimos la categoria y buscamos los gifs de esa categoria
   
   return (
      <>
         <h3 className="animate__animated animate__fadeIn">{category}</h3>
         {loading && <p className="animate__animated animate__flash">Cargando...</p>}
         <div className="card-grid">
            {
               images.map(img => {
                  // return <li key={id}>{title}</li>
                  return <GifGridItem
                     key={img.id}
                     {...img}
                  />
               })
            }

         </div>
      </>
   )
}

GifGrid.propTypes = {
   category:PropTypes.string
}

export default GifGrid
