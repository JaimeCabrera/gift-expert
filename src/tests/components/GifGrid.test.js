import React from 'react';

import { shallow } from "enzyme";
import { GifGrid } from "../../components/GifGrid";
import '@testing-library/jest-dom';
import { useFetchGifs } from '../../hooks/useFetchGifs';

// fingir la lamada al archivo usefetchgiufs
jest.mock('../../hooks/useFetchGifs');

describe('Pruebas en el componente <GifGrid/>', () => {

   const category = "futurama";
   test('debe cargar correctamente el componente', () => {

      useFetchGifs.mockReturnValue({
         data: [],
         loading: true
      });
      const wrapper = shallow(<GifGrid category={category} />);
      expect(wrapper).toMatchSnapshot();
   });

   test('Debde de mostrar items cuando se cargan las imagens, useFetchGifs', () => {

      const gifs = [{
         id: '1',
         title: 'titulo del gif',
         url: 'http://gifs/gif.jpg'
      }];

      useFetchGifs.mockReturnValue({
         data: gifs,
         loading: false
      });
      // hacer un mock fingir algo ->usamos el userFetchGifs el import pero no lo usamos
      const wrapper = shallow(<GifGrid category={category} />);
      // asercion que es lo que necesito evaluar
      expect(wrapper).toMatchSnapshot();
      // exists() no deberia existri
      expect(wrapper.find('p').exists()).toBe(false);
      expect(wrapper.find('GifGridItem').length).toBe(gifs.length)

   })



})
