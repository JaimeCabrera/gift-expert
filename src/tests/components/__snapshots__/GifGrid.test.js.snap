// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <GifGrid/> Debde de mostrar items cuando se cargan las imagens, useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    futurama
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="1"
      key="1"
      title="titulo del gif"
      url="http://gifs/gif.jpg"
    />
  </div>
</Fragment>
`;

exports[`Pruebas en el componente <GifGrid/> debe cargar correctamente el componente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    futurama
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Cargando...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
